cpp/tests/io/parquet_test.cpp

```diff
@@ -2870,7 +2870,7 @@ TEST_F(ParquetReaderTest, ReorderedColumns)
     md.column_metadata[0].set_name("a");
     md.column_metadata[1].set_name("b");
     cudf::io::parquet_writer_options opts =
-      cudf::io::parquet_writer_options::builder(cudf::io::sink_info{filepath}, tbl).metadata(&md);
+      cudf::io::parquet_writer_options::builder(cudf::io::sink_info{filepath}, tbl).metadata(md);
```

Comment:
Some of these changed to `std::move(md)` and some are just `md`. Is there reasoning behind this? Should they all be moved? If so, maybe we should enforce that by taking an rvalue reference.